clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1994,6 +1995,24 @@ public void testGetGroupMetadata() {
         assertEquals(groupInstanceId, groupMetadataAfterPoll.groupInstanceId());
     }
 
+    @Test
+    public void testInvalidGroupMetadata() {
+        Properties props = new Properties();
+        props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        props.setProperty(ConsumerConfig.METRIC_REPORTER_CLASSES_CONFIG, MockMetricsReporter.class.getName());
+        KafkaConsumer<String, String> consumer = new KafkaConsumer<>(
+                props, new StringDeserializer(), new StringDeserializer());
+
+        // concurrent access is illegal
+        consumer.currentThread.set(123);
```

Comment:
Another way to test this might be to use `MockClient.enableBlockingUntilWakeup`. That would let us block in `Consumer.poll`.

Reply:
will Roger that!